pkg/monitor/agent.go

```diff
@@ -89,15 +100,15 @@ func (monitor *agentMonitor) createAgentCRD() (*v1beta1.AntreaAgentInfo, error)
 }
 
 // updateAgentCRD updates all the fields of existing monitoring CRD.
-func (monitor *agentMonitor) updateAgentCRD(agentCRD *v1beta1.AntreaAgentInfo) (*v1beta1.AntreaAgentInfo, error) {
-	monitor.querier.GetAgentInfo(agentCRD, false)
-	klog.V(2).Infof("Updating agent monitoring CRD %+v", agentCRD)
-	return monitor.client.ClusterinformationV1beta1().AntreaAgentInfos().Update(agentCRD)
+func (monitor *agentMonitor) entireUpdateAgentCRD() (*v1beta1.AntreaAgentInfo, error) {
+	monitor.querier.GetAgentInfo(monitor.agentCRD, false)
+	klog.V(2).Infof("Updating agent monitoring CRD %+v", monitor.agentCRD)
+	return monitor.client.ClusterinformationV1beta1().AntreaAgentInfos().Update(monitor.agentCRD)
 }
 
 // partialUpdateAgentCRD only updates some variables.
-func (monitor *agentMonitor) partialUpdateAgentCRD(agentCRD *v1beta1.AntreaAgentInfo) (*v1beta1.AntreaAgentInfo, error) {
-	monitor.querier.GetAgentInfo(agentCRD, true)
-	klog.V(2).Infof("Partially updating agent monitoring CRD %+v", agentCRD)
-	return monitor.client.ClusterinformationV1beta1().AntreaAgentInfos().Update(agentCRD)
+func (monitor *agentMonitor) partialUpdateAgentCRD() (*v1beta1.AntreaAgentInfo, error) {
```

Comment:
Maybe it could be merged to updateAgentCRD:
```
func (monitor *agentMonitor) updateAgentCRD(agentCRD *v1beta1.AntreaAgentInfo, partial bool) (*v1beta1.AntreaAgentInfo, error) {
	monitor.querier.GetAgentInfo(agentCRD, partial)
	klog.V(2).Infof("Updating agent monitoring CRD %+v, partial: %t", agentCRD, partial)
	return monitor.client.ClusterinformationV1beta1().AntreaAgentInfos().Update(agentCRD)
}
```

Reply:
Good point, have merged them to one.